src/mscorlib/src/System/Collections/Generic/EqualityComparer.cs

```diff
@@ -61,7 +61,7 @@ private static EqualityComparer<T> CreateComparer()
             }
             // See the METHOD__JIT_HELPERS__UNSAFE_ENUM_CAST and METHOD__JIT_HELPERS__UNSAFE_ENUM_CAST_LONG cases in getILIntrinsicImplementation
             else if (t.IsEnum) {
-                TypeCode underlyingTypeCode = Type.GetTypeCode(Enum.GetUnderlyingType(t));
+                TypeCode underlyingTypeCode = Type.GetTypeCode(t);
```

Comment:
Can you double check that you are actually going to get the right TypeCode after this change?